bookShop/src/app/core/services/cart.service.ts

```diff
@@ -22,8 +23,11 @@ export class CartService {
 
   totalSum = 0;
 
-  constructor(private booksService: BooksService) {
-    this.booksService.getBooks().subscribe((n) => (this.booksData = n));
+  constructor(
+    private booksService: BooksService,
+    private localStorageService: LocalStorageService
+  ) {
+    this.booksService.getBooks().subscribe((data) => (this.booksData = data));
```

Comment:
It's a bad idea to subscribe inside the constructor. Don't forget to unsubscribe.

Reply:
Fixed